SwiftGit2/Errors.swift

```diff
@@ -1,50 +1,193 @@
+
+	/// More than one object matches
+	case multipleObjectsExist(libgit2Error: Libgit2Error)
+
+	/// Output buffer too short to hold data
+	case outputBufferTooShort(libgit2Error: Libgit2Error)
+
+	/// a special error that is never generated by libgit2
+	/// code.  You can return it from a callback (e.g to stop an iteration)
+	/// to know that it was generated by the callback and not by libgit2.
+	case userContext(libgit2Error: Libgit2Error)
+
+	/// Operation not allowed on bare repository
+	case bareRepo(libgit2Error: Libgit2Error)
+
+	/// HEAD refers to branch with no commits
+	case headWithoutCommits(libgit2Error: Libgit2Error)
+
+	/// Merge in progress prevented operation
+	case mergeInProgress(libgit2Error: Libgit2Error)
+
+	/// Reference was not fast-forwardable
+	case referenceNotFastForwardable(libgit2Error: Libgit2Error)
+
+	/// Name/ref spec was not in a valid format
+	case invalidSpec(libgit2Error: Libgit2Error)
+
+	/// Checkout conflicts prevented operation
+	case checkoutConflicts(libgit2Error: Libgit2Error)
+
+	/// Lock file prevented operation
+	case fileLocked(libgit2Error: Libgit2Error)
+
+	/// Reference value does not match expected
+	case unexpectedReferenceValue(libgit2Error: Libgit2Error)
+
+	/// Authentication error
+	case authenticationError(libgit2Error: Libgit2Error)
+
+	/// Server certificate is invalid
+	case invalidServerCertificate(libgit2Error: Libgit2Error)
 
-		if let pointOfFailure = pointOfFailure {
-			userInfo[NSLocalizedFailureReasonErrorKey] = "\(pointOfFailure) failed."
+	/// Patch/merge has already been applied
+	case alreadyApplied(libgit2Error: Libgit2Error)
+
+	/// The requested peel operation is not possible
+	case peelNotPossible(libgit2Error: Libgit2Error)
+
+	/// Unexpected EOF
+	case endOfFile(libgit2Error: Libgit2Error)
+
+	/// Invalid operation or input
+	case invalidOperation(libgit2Error: Libgit2Error)
+
+	/// Uncommitted changes in index prevented operation
+	case uncommitedChanges(libgit2Error: Libgit2Error)
+
+	/// The operation is not valid for a directory
+	case invalidDirectory(libgit2Error: Libgit2Error)
+
+	/// A merge conflict exists and cannot continue
+	case mergeConflict(libgit2Error: Libgit2Error)
+
+	/// Signals end of iteration with iterator
+	case endOfIteration(libgit2Error: Libgit2Error)
+
+	/// Hashsum mismatch in object
+	case hashMismatch(libgit2Error: Libgit2Error)
+
+	/// Object has an unrecognized type
+	case unknownObjectType(git_otype, OID)
+
+	/// Could not compute diff
+	case diffError
+
+	public var description: String {
+		if let underlyingError = underlyingError {
+			return underlyingError.description
+		}
+		switch self {
+		case .unknownObjectType(let type, let oid):
+			return "Unrecognized git_otype '\(type)' for oid '\(oid)'."
+		case .diffError:
+			return "Diff Error"
+		default:
+			return "Unknown"
 		}
+	}
 
-		self.init(domain: libGit2ErrorDomain, code: code, userInfo: userInfo)
+	internal init(libgitErrorCode code: Int32, source: Libgit2Method) {
```

Comment:
Can you add the documentation from above here again? To make clear what these params are doing. Feel free to adjust the text below.
```
        /// Returns an SwiftGit2Error with an error code and message for libgit2 errors.
 	///
 	/// :param: code An error code returned by a libgit2 function.
 	/// :param: source The name of the libgit2 method that produced the
 	///         error code.
 	/// :returns: An SwiftGit2Error with a libgit2 error code and the corresponding source message.
```